front/src/applications/stdcm/components/StdcmResults/StdcmResults.tsx

```diff
@@ -93,6 +96,46 @@ const StdcmResults = ({
     return extractMarkersInfo(outputs.results.simulationPathSteps);
   }, [hasSimulationResults, outputs]);
 
+  const [refSchedule, setRefSchedules] = useState<string[]>([]);
+
+  const [postRefSchedules] = osrdEditoastApi.endpoints.postSimilarTrains.useMutation();
+  const requestTrainSchedules = async () => {
+    const { consist, pathSteps } = selectedSimulation.inputs;
+    if (!consist) {
+      return;
+    }
+    const key = (uic: number, ch: string) => `${uic}-${ch}`;
+    const stops = pathSteps.reduce((acc, ps) => {
+      const k = key(ps.location?.uic ?? -1, ps.location?.secondary_code ?? '');
+      acc.set(k, !ps.isVia || (ps.stopFor !== undefined && ps.stopFor.ms > 0));
+      return acc;
+    }, new Map<string, boolean>());
+    const waypoints =
+      selectedSimulation.outputs?.pathProperties.manchetteOperationalPoints?.map((op) => {
+        const k = key(op.extensions?.identifier?.uic ?? -1, op.extensions?.sncf?.ch ?? '');
```

Comment:
Hum why use -1 if the uic is undefined ?

Reply:
because the endpoint needs a number